src/InstagramClone.Api/Entities/User_UserFollower.cs

```diff
@@ -0,0 +1,11 @@
+﻿namespace InstagramClone.Api.Entities
+{
+    public class User_UserFollower
+    {
+        public User User { get; set; }
+        public Guid Id { get; set; }
+        public UserFollower UserFollower { get; set; }
+        public Guid FollowerId { get; set; }
+
+    }
+}
```

Comment:
Why have you introduced this class and what does it do?

Reply:
From the material I used this is a join table to join entity to reference the navigation and Fk properties to configure a many-to-many relationship

Reply:
Good. So how is it different from the already existing `UserFollower` class?

Reply:
So now I realized I have two duplicate tables.  so now I am confused, Is it that there are two methods to do this, or was I supposed to leave the 1st table and do the two one-to-many relationships?

Reply:
so this is good. I wanted you to realise it for yourself that it's a duplicate. 👌 

I like the direction you're going with the many to many, but you didn't need another table/class for it. The UserFollower class is sufficient.

So here's the thing. For the many to may relationship, A `User` SHOULD have 2 key relationships 

- 1 `User` to  **many** "followers"`UserFollower` and 
- 1 `User` to **many** "following" `UserFollower`, 

so First, I'll expect these 2 (Followers and Following) to be defined as Navigation properties on the `User` class. Note the plural

So that means you'll have 2 separate **1 to many** relationships configured from `User` to `UserFollower`, one for each Navigation property.

Each of these relationships will have to be linked to the `Follower` and `Followed` Navigation properties on the `UserFollower` class. That's the key to nailing this.

Read through this and re-read that article and let me know if it is clear.